tests/components/airpatrol/test_coordinator.py

```diff
@@ -0,0 +1,133 @@
+"""Test the AirPatrol data update coordinator."""
+
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from airpatrol.api import AirPatrolAuthenticationError, AirPatrolError
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant, State
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_update_data_with_stored_token(
+    hass: HomeAssistant,
+    mock_api_response: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    get_data,
+) -> None:
+    """Test data update with stored access token."""
+    mock_api_response.return_value = get_data()
+    mock_config_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    mock_api_response.assert_called()
+    assert mock_api_response.call_count == 2
+    state: State | None = hass.states.get("climate.living_room")
+    assert state
+    assert state.state == "cool"
+    assert state.attributes["temperature"] == 22.0
+    assert state.attributes["current_temperature"] == 22.5
+    assert state.attributes["current_humidity"] == 45.0
```

Comment:
I am nto sure what this tests other than fetching data what the climate tests already test